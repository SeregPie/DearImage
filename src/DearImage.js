export default class {
	static is(value) {
		return value instanceof this;
	}
	constructor(context) {
		this.context = context;
	}
	get canvas() {
		return this.context.canvas;
	}
	get width() {
		return this.canvas.width;
	}
	get height() {
		return this.canvas.height;
	}
	get sizeX() {
		return this.width;
	}
	get sizeY() {
		return this.height;
	}
}
