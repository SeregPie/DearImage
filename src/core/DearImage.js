export default class {
	static is(value) {
		return value instanceof this;
	}
	constructor(context) {
		this.context = context;
	}
	get canvas() {
		return this.context.canvas;
	}
	get sizeX() {
		return this.canvas.width;
	}
	get sizeY() {
		return this.canvas.height;
	}
}
