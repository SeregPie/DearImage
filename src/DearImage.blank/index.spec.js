import DearImage from '../DearImage';

import './';

describe('DearImage.blank', () => {
	test('lqzsubmjqarh', () => {
		let sizeX = 300;
		let sizeY = 150;
		let image = DearImage.blank(sizeX, sizeY);
		expect(image.sizeX).toBe(sizeX);
		expect(image.sizeY).toBe(sizeY);
	});
	test('gitsmncelnqe', () => {
		let image = DearImage.blank();
		expect(image.sizeX).toBe(0);
		expect(image.sizeY).toBe(0);
	});
});
