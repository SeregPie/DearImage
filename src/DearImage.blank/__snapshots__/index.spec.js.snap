// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DearImage.blank should work 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAICAYAAADwdn+XAAAABmJLR0QA/wD/AP+gvaeTAAAAD0lEQVQokWNgGAWjgIEBAAIIAAHQKWGuAAAAAElFTkSuQmCC"`;
